### ISSUE TITLE
Rails/ApplicationRecord not safe?

### ISSUE THREAD

**Issue by vfonic** [rubocop-hq/rubocop#6898]
I ran `rubocop --safe-auto-correct` in my app and my app broke.

The issue was with this gem:
https://github.com/infinum/phrasing
As you can see [here](https://github.com/infinum/phrasing/blob/master/app/models/phrasing_phrase.rb#L1), `PhrasingPhrase` inherits from `ActiveRecord::Base`.

I'm using this gem in one of my applications. I needed to "open" the class and make modifications, so I wrote a code similar to this:

```ruby
class PhrasingPhrase < ActiveRecord::Base
  # my code here
end
```

After running `rubocop --safe-auto-correct`, the above code changed to:

```ruby
class PhrasingPhrase < ApplicationRecord
  # my code here
end
```

This broke my application, as the class is no longer inheriting form the same parent class and that's a runtime error.

--------

## Expected behavior

I'd expect this cop to autocorrect this error only for `rubocop --auto-correct` and no `rubocop --safe-auto-correct`. => I'd expect this cop to be marked as not safe.

## Actual behavior

Runtime error.

## RuboCop version

```
$ bundle exec rubocop -V
0.67.2 (using Parser 2.6.2.1, running on ruby 2.5.3 x86_64-darwin17)
```

**Comment by vfonic** [comment]
Bump! Any pair of eyes want to check this out? Thank you!

**Comment by bbatsov** [comment]
See https://github.com/rubocop-hq/rubocop/pull/6889

**Comment by vfonic** [comment]
@bbatsov I can't seem to find the connection with that PR and this issue?

Is it this line mentioning adding `ActiveRecord::Base`?
https://github.com/rubocop-hq/rubocop/pull/6889#discussion_r272660253

This is a different cop than the one in linked PR.

Or are you trying to say that most of the rails cops could be marked as unsafe, so your decision is to not mark them for such "edge cases"? :)

**Comment by bbatsov** [comment]
Ah, my bad. But yeah - in general most Rails cops are unsafe, as they rely on imprecise heuristics to operate (mostly some naming conventions). At some point we'll have to decide if we want to be strict and make them unsafe explicitly. Probably we should.

**Comment by vfonic** [comment]
I agree. I'd love to be able to confidently run `--safe-auto-correct` and then, after I commit those autofixes, without even blinking, I can run `--auto-correct` and check all the autofixes one by one to decide which ones are safe and I'll keep. :+1:

This could actually be a suggested way that rubocop can show to users of the tool.

I'll close this issue for now as it's a part of a bigger issue to tackle. Thanks for answering!
